cli/src/commands/describe.rs

```diff
@@ -67,35 +77,140 @@ pub(crate) fn cmd_describe(
     args: &DescribeArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let commit = workspace_command.resolve_single_rev(&args.revision)?;
-    workspace_command.check_rewritable([commit.id()])?;
-    let description = if args.stdin {
+    let commits: Vec<_> = workspace_command
+        .parse_union_revsets(&args.revisions)?
+        .evaluate_to_commits()?
+        .try_collect()?; // in reverse topological order
+    if commits.is_empty() {
+        writeln!(ui.status(), "No revisions to describe.")?;
+        return Ok(());
+    }
+    workspace_command.check_rewritable(commits.iter().ids())?;
+
+    let shared_description = if args.stdin {
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer).unwrap();
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        let template =
-            description_template_for_describe(ui, command.settings(), &workspace_command, &commit)?;
-        edit_description(workspace_command.repo(), &template, command.settings())?
+        None
     };
-    if description == *commit.description() && !args.reset_author {
-        writeln!(ui.status(), "Nothing changed.")?;
+    let commit_descriptions: HashMap<_, _> = if args.no_edit || shared_description.is_some() {
+        commits
+            .iter()
+            .rev()
+            .filter_map(|commit| -> Option<(&CommitId, String)> {
+                let original_description = commit.description();
+                let new_description = if let Some(shared_description) = &shared_description {
+                    shared_description
+                } else {
+                    original_description
+                };
+                if *new_description == *original_description && !args.reset_author {
```

Comment:
nit: perhaps, this filtering can be factored out? I don't think the cost of extra hash map insertion would matter.

Reply:
The intent here is to avoid rewriting the commit unless the description has changed or `--reset-author` was set. It can be done either before the rewriting (which is in both branches of the if-else here), or later on in the `transform_descendants` callback.

Reply:
Yeah, I just meant `commit_descriptions` (or commit ids passed to `transform_descendants()`) can be filtered later to deduplicate `.filter_map()`s.